### PR TITLE
ENH: Scale dependent slope

### PR DESCRIPTION
**Pull request by pastewka**
This closes issue #403

### CODE REVIEW

**Review by mcrot — CHANGES_REQUESTED** [review]
Looks good. Two proposals for change:

- so far, we used unit "1" for slopes not None, I think this should be consistent, I'm also not sure how the plotting behaves (what is displayed as unit there?), so I suggest to also use "1"
- There is one comment which is no longer applicable, I will mark it here

File: topobank/analysis/functions.py [hunk 1]
```diff
@@ -716,6 +716,43 @@ def autocorrelation_for_surface(surface, num_points=100,
...
+                                  storage_prefix=storage_prefix)
+
+    for dataset in return_dict['series']:
+        x = dataset['x']
+        y = dataset['y']
+        dataset['x'] = x[1:]
+        dataset['y'] = np.sqrt(2 * y[1:]) / x[1:]
+    return_dict['name'] = 'Scale-dependent slope'
+    return_dict['ylabel'] = 'Slope'
+    return_dict['yunit'] = None
```
Comment (mcrot) [inline topobank/analysis/functions.py:732]: I suggest to use "1" as "yunit".

File: topobank/analysis/tests/test_functions.py [hunk 2]
```diff
@@ -357,7 +358,7 @@ def test_curvature_distribution_simple_2d_topography_periodic():
     arr = np.sin(y / 2 / np.pi)  # only slope in y direction, second derivative is -sin
 
     t = Topography(arr, (100, 100), periodic=True, info=info).detrend('center')
-    # resulting heights follow this function: h(x,y)=-4y+9
```
Comment (mcrot) [inline topobank/analysis/tests/test_functions.py:360]: Thanks for fixing this.

File: topobank/analysis/tests/test_functions.py [hunk 3]
```diff
@@ -719,6 +733,85 @@ def test_autocorrelation_for_surface(mocker):
...
+def test_scale_dependent_slope_for_surface(mocker):
+    """Testing autocorrelation for an artificial surface."""
+
+    # ACF results for individual topographies are mocked up
+    topo1_result = dict(
+        name='Scale-dependent Slope',
+        xlabel='Distance',
+        ylabel='Slope',
+        xunit='µm',
+        yunit=None,
```
Comment (mcrot) [inline topobank/analysis/tests/test_functions.py:746]: Also "1"?

File: topobank/analysis/tests/test_functions.py [hunk 4]
```diff
@@ -719,6 +733,85 @@ def test_autocorrelation_for_surface(mocker):
...
+        xscale='log',
+        yscale='log',
+        series=[
+            {
+                'name': '1D',  # taken from y=x
+                'x': np.array([1, 2, 3, 5, 6, 7]),
+                'y': np.array([1, 2, 3, 5, 6, 7]),
+                # using here 0.9 instead of 1 so we can clearly
```
Comment (mcrot) [inline topobank/analysis/tests/test_functions.py:754]: This comment with 0.9 is not valid here any more.

File: topobank/analysis/tests/test_functions.py [hunk 5]
```diff
@@ -719,6 +733,85 @@ def test_autocorrelation_for_surface(mocker):
...
+    result = scale_dependent_slope_for_surface(surf, num_points=15)
+
+    expected_x, expected_y, expected_std_err_y = _expected('log')
+
+    expected_result = {
+        'name': 'Scale-dependent Slope',
+        'xlabel': 'Distance',
+        'ylabel': 'Slope',
+        'xunit': 'µm',
+        'yunit': None,
```
Comment (mcrot) [inline topobank/analysis/tests/test_functions.py:793]: Also "1"

**Comment by pastewka** [comment]
I've made all requested changes

**Pull request merged**